AdobeStockImageAdminUi/Test/Mftf/Test/AdminAdobeStockNotPreviewedLicensedImageSaveTest.xml

```diff
@@ -0,0 +1,51 @@
+            <stories value="User saves licensed image into Magento Media Gallery"/>
+            <title value="Adobe Stock Not Previewed Licensed Image Save"/>
+            <description value="User saves previously licensed image into Magento Media Gallery"/>
+            <severity value="CRITICAL"/>
+            <group value="adobe_stock_integration_license"/>
+            <group value="adobe_stock_integration"/>
+        </annotations>
+        <before>
+            <actionGroup ref="LoginAsAdmin" stepKey="loginAsAdmin"/>
+            <actionGroup ref="AdminOpenMediaGalleryForPageNoEditorActionGroup" stepKey="openMediaGalleryForPage"/>
+            <actionGroup ref="AdminAdobeStockOpenPanelFromMediaGalleryActionGroup" stepKey="openAdobeStockPanel"/>
+            <actionGroup ref="AdminAdobeStockClickSignInActionGroup" stepKey="clickOnSignIn"/>
+            <actionGroup ref="AdminAdobeStockImsPopupSignInFillUserDataActionGroup" stepKey="fillUserCredentials"/>
+            <actionGroup ref="AdminAdobeStockImsPopupClickSignInActionGroup" stepKey="clickSignInImsPopup"/>
+            <actionGroup ref="AdminAdobeStockAssertUserLoggedActionGroup" stepKey="assertUserLoggedIn"/>
+        </before>
+        <after>
+            <actionGroup ref="AdminOpenMediaGalleryForPageNoEditorActionGroup" stepKey="openMediaGalleryForPage"/>
+            <actionGroup ref="AdminMediaGalleryDeleteImage" stepKey="deleteImageActionGroup">
+                <argument name="generatedImageName" value="{$grabImageFileName}"/>
+            </actionGroup>
+            <actionGroup ref="AdminAdobeStockOpenPanelFromMediaGalleryActionGroup" stepKey="openAdobeStockPanel"/>
+            <actionGroup ref="AdminAdobeStockUserSignOutActionGroup" stepKey="adobeLogout"/>
+            <actionGroup ref="AdminAdobeStockAssertUserNotLoggedActionGroup" stepKey="assertUserLoggedOut"/>
+            <actionGroup ref="resetAdminDataGridToDefaultView" stepKey="resetAdminDataGridToDefaultView"/>
+            <actionGroup ref="logout" stepKey="adminLogout"/>
+        </after>
+        <actionGroup ref="AdminSearchImagesOnModalActionGroup" stepKey="searchForNotPreviewedLicensedImageToSave">
+            <argument name="query" value="{{AdobeStockConfigDataNotPreviewedLicensedNotSavedImage.value}}"/>
+        </actionGroup>
+        <actionGroup ref="AdminAdobeStockExpandImagePreviewActionGroup" stepKey="expandLicensedImage"/>
+        <click selector="{{AdobeStockImagePreviewSection.saveLicensedImage}}" stepKey="clickSave"/>
+        <waitForPageLoad stepKey="waitForPromptModal" />
+        <grabValueFrom selector="{{AdobeStockImagePreviewSection.generatedImageName}}" stepKey="grabImageFileName" />
+        <click selector="{{AdobeStockImagePreviewSection.confirm}}" stepKey="clickOnPopupConfirm"/>
+        <waitForPageLoad stepKey="waitForMediaGalleryOpen" />
+        <seeElement selector="{{AdobeStockSection.mediaGalleryImage({$grabImageFileName})}}" stepKey="assertSavedImage" />
```

Comment:
@diazwatson I used direct code here instead of the action group: `AdminAssertAdobeStockImageSavedActionGroup` as this test and the other test in this PR have the same selected actions, but in a slightly different order. So I'm wondering if I can decouple that action group, and reorder as necessary.

Reply:
This is still an issue regarding the DRY principle. I can refactor `AdminAssertAdobeStockImageSavedActionGroup` into two action groups, and reuse them here, IF it is possible to pass `stepKey` data from one action group to another (need to pass image file name from the first action group to the other). So the above code would become:
```
<!-- The decoupled action groups -->
<actionGroup ref="AdminAdobeStockGrabImageNameActionGroup" name="grabImageName"/>
<actionGroup ref="AdminAssertAdobeStockImageSavedActionGroup" name="assertImageSave"/>
```
instead of 
```
<!-- The coupled action groups -->
<actionGroup ref="AdminAssertAdobeStockImageSavedActionGroup" name="assertImageSave"/>
```

Reply:
I would suggest making the mentioned actions. Instead of using the direct actions in the test, it's better to create two action groups:
- `AdminAdobeStockGrabImageNameForSaveActionGroup`
- `AdminAssertImageInMediaGalleryActionGroup`

Please note the name of the second action group. It's not something related to AdobeStock, but the standard media gallery. There's a chance that something similar already exists, I would recommend to double-check it.

Reply:
Thanks, @rogyar! I shall implement these changes.